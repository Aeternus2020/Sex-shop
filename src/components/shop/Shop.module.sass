img
    width: 250px
    border-radius: 10px

h1
    text-align: center;

li
    list-style-type: none
    flex: 2 300px
    width: 300px
    max-height: 400px
    max-width: 300px
    display: flex
    flex-direction: column
    align-items: center
    gap: 10px
    border-radius: 20px
    border: solid Silver 2px
    padding: 10px
    -webkit-box-shadow: 0px 0px 30px 6px rgba(66, 68, 90, 1)
    -moz-box-shadow: 0px 0px 30px 6px rgba(66, 68, 90, 1)
    box-shadow: 0px 0px 30px 6px rgba(66, 68, 90, 1)
    button
        background-color: DimGray
        color: white
        padding: 5px
        font-size: 1.1rem
        cursor: pointer

ul
    display: flex
    flex-flow: row wrap
    justify-content: center
    max-width: 1200px
    gap: 50px
    margin: 50px auto

.name
    font: 1.5rem bold
    margin: 5px

.price
    display: flex
    align-items: center
    font-size: 1.3rem

.btnModal
    background-color: black
    color: white

.shop
    display: flex
    justify-content: flex-start
    max-width: 1300px
    margin: 50px auto

.style
    display: flex
    align-items: flex-start
    position: relative
    left: 10%
    button
        background-color: DimGray
        color: white
        padding: 5px
        font-size: 1.3rem
        width: 70px
        cursor: pointer

.list
    img
        display: none
    li
        flex-direction: row
        flex: 1 1 100px
        height: 70px
    button
        font-size: 1.1rem
        position: static
    