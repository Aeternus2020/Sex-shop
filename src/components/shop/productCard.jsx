import styles from "../shop/Shop.module.sass";
import { useSelector } from "react-redux";
import ButtonModal from "../buttonModal";
import StarSolid from "../../pages/favorite/star";
import { useState } from "react";

export function ProductCard(props) {
    const [star, setStar] = useState('#000');
    const countFavorite = useSelector(state => state.favorite.favoriteArray);
    const buttonCross = useSelector(state => state.showModal.button);

    return (
        <li key={props.item.article}  data-testid={`table-row-${props.id}`}>
        <div className={`${styles.name}`}>{props.item.name}</div>
        <img src={props.item.url} alt={props.item.name} /> 
        <div className={`${styles.price}`}>
            <div>Price: {props.item.price}</div>
            <StarSolid 
                favoriteArray={countFavorite} 
                item={props.item} 
                star={star}/>
        </div>
        {buttonCross && 
            <ButtonModal 
            className = {props.className}
            text={props.text}
            item={props.item} 
            textbtn={props.textbtn}/>}
        </li>
    )
}