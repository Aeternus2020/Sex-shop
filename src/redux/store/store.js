import { applyMiddleware, combineReducers, createStore } from "redux";
import products from "../getProducts/reducers";
import basket from "../pages/basket/reducers";
import favorite from "../pages/favorite/reducers";
import showModal from "../components/modal/reducers"; 
import title from "../pages/app/reducers";
import { composeEnhancers, middleware } from "./middleware";

export const rootReducer = combineReducers({
    products,
    basket,
    favorite,
    showModal,
    title
})

export const store = createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(...middleware))
    )