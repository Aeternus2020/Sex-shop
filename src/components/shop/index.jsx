import React, {useEffect, useContext } from 'react';
import styles from "../shop/Shop.module.sass";
import { useDispatch, useSelector } from 'react-redux';
import { getProductsAsync } from '../../redux/getProducts/actions'; 
import { ShopContext } from '../../App';
import { ProductCard } from './productCard';

export function Shop(props)  {
  const dispatch = useDispatch();
  const products = useSelector((state) => state.products);
  const buttonStyle = useSelector(state => state.showModal.button);
  const [style, setStyle] = useContext(ShopContext);

  function changeStyleShop(item) {
    setStyle(item)
  }
  
    useEffect(() => {
        dispatch(getProductsAsync(props.productsFile))
    }, [dispatch]);
    
      return (
        <>
        {buttonStyle && 
        <div className={`${styles.style}`}>
          <button onClick={()=>changeStyleShop('list')}>List</button>
          <button onClick={()=>changeStyleShop('')}>Table</button>
        </div>}
          <ul className={`${styles[style]}`}>
          {products?.map((item) => (
            <ProductCard 
            key={item.article}
            item={item}
            className = {props.className}
            text={props.text}
            textbtn={props.textbtn}/>
          ))}
        </ul>
      </>
      );
    }

export default Shop;