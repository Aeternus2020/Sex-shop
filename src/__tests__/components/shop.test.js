import { ProductCard } from "../../components/shop/productCard";
import Shop from "../../components/shop";
import { render, screen, cleanup } from "@testing-library/react";
import { store } from "../../redux/store/store";
import { Provider } from "react-redux";
import renderer from 'react-test-renderer';

afterEach(cleanup);

test('Test basic props in ProductCard', () => {
    const item = {
        name: "Bracelets",
        price: "120$",
        url: "./images/bracelets.webp",
    }
    render(
        <Provider store={store}>
        <ProductCard item={item} id={1}/>
    </Provider>
        );
    expect(screen.getByTestId('table-row-1')).toBeInTheDocument();
    expect(screen.getByTestId('table-row-1')).toHaveTextContent('Bracelets');
    expect(screen.getByTestId('table-row-1')).toHaveTextContent('120$');
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.getByRole('button')).toBeInTheDocument();
})

test('Test basic props in Shop', () => {
    render(
    <Provider store={store}>
        <Shop id={2}/>
    </Provider>
        );
    expect(screen.getAllByRole('button')).toHaveLength(2);
    expect(screen.getAllByRole('button')[0]).toHaveTextContent('List');
    expect(screen.getAllByRole('button')[1]).toHaveTextContent('Table');
    expect(screen.getByTestId('buttons-style-2')).toBeInTheDocument();
    expect(screen.getByTestId('card-list-2')).toBeInTheDocument();
})

test('Test snapshot Shop', () => {
    const tree = renderer.create(
    <Provider store={store} >
        <Shop id={2}/>
    </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})