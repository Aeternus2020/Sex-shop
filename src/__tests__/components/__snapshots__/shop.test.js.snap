// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Shop 1`] = `
Array [
  <div
    className="style"
    data-testid="buttons-style-2"
  >
    <button
      onClick={[Function]}
    >
      List
    </button>
    <button
      onClick={[Function]}
    >
      Table
    </button>
  </div>,
  <ul
    className="table"
    data-testid="card-list-2"
  />,
]
`;
